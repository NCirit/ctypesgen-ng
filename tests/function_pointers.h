typedef int (*compar_t)(const void *, const void *);

extern compar_t default_cmp;

void my_qsort(void *base, unsigned long nmemb, unsigned long size, compar_t compar);

struct vtable {
	compar_t cmp;
	void *(*add)(const void *, const void *);
};

typedef void (*meta_t)(void (*)(void));

void (*signal(int signum, void (*handler)(int)))(int);

typedef int (*noproto_t)();

typedef (*defaultrettype_t)(char *);

typedef (*blasphemy_t)();

#if 0
/*
 * TODO: not sure why the compar_t parameter comes up as a function pointer and
 * not a typedef.
 */
typedef void (*wrapper_t)(compar_t);
#endif
