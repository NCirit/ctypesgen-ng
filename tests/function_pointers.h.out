import ctypes

libtest = ctypes.CDLL('libtest.so')

typedef_compar_t = ctypes.CFUNCTYPE(ctypes.c_int, ctypes.c_void_p, ctypes.c_void_p)

var_default_cmp = typedef_compar_t.in_dll(libtest, 'default_cmp')

libtest.my_qsort.restype = None
libtest.my_qsort.argtypes = [ctypes.c_void_p, ctypes.c_ulong, ctypes.c_ulong, typedef_compar_t]


class struct_vtable(ctypes.Structure):
    pass
struct_vtable._fields_ = [
    ('cmp', typedef_compar_t),
    ('add', ctypes.CFUNCTYPE(ctypes.c_void_p, ctypes.c_void_p, ctypes.c_void_p)),
]

typedef_meta_t = ctypes.CFUNCTYPE(None, ctypes.CFUNCTYPE(None))

libtest.signal.restype = ctypes.CFUNCTYPE(None, ctypes.c_int)
libtest.signal.argtypes = [ctypes.c_int, ctypes.CFUNCTYPE(None, ctypes.c_int)]

typedef_noproto_t = ctypes.CFUNCTYPE(ctypes.c_int)

typedef_defaultrettype_t = ctypes.CFUNCTYPE(ctypes.c_int, ctypes.c_char_p)

typedef_blasphemy_t = ctypes.CFUNCTYPE(ctypes.c_int)
